src/families/cosmos/bridge/libcore.js

```diff
@@ -0,0 +1,237 @@
+// @flow
+import { scanAccounts } from "../../../libcore/scanAccounts";
+import { sync } from "../../../libcore/syncAccount";
+import type { AccountBridge, CurrencyBridge } from "../../../types";
+import type { Transaction } from "../types";
+import { BigNumber } from "bignumber.js";
+import broadcast from "../libcore-broadcast";
+import signOperation from "../libcore-signOperation";
+import { getMainAccount } from "../../../account";
+import { makeLRUCache } from "../../../cache";
+import { validateRecipient } from "../../../bridge/shared";
+import { getFeesForTransaction } from "../../../libcore/getFeesForTransaction";
+import {
+  NotEnoughBalance,
+  InvalidAddressBecauseDestinationIsAlsoSource,
+  InvalidAddress,
+} from "@ledgerhq/errors";
+import {
+  CosmosTooMuchRedelegations,
+  CosmosTooMuchUnboundings,
+  CosmosRedelegationInProgress,
+} from "../../../errors";
+import {
+  setCosmosPreloadData,
+  asSafeCosmosPreloadData,
+} from "../preloadedData";
+import { getValidators, hydrateValidators } from "../validators";
+
+const calculateFees = makeLRUCache(
+  async (a, t) => {
+    return getFeesForTransaction({
+      account: a,
+      transaction: t,
+    });
+  },
+  (a, t) =>
+    `${a.id}_${t.amount.toString()}_${t.recipient}_${
+      t.gasLimit ? t.gasLimit.toString() : ""
+    }_${t.fees ? t.fees.toString() : ""}
+    _${String(t.useAllAmount)}_${t.mode}_${
+      t.validators ? t.validators.map((v) => v.address).join("-") : ""
+    }`
+);
+
+const createTransaction = () => ({
+  family: "cosmos",
+  mode: "send",
+  amount: BigNumber(0),
+  fees: null,
+  gasLimit: null,
+  recipient: "",
+  useAllAmount: false,
+  networkInfo: null,
+  memo: null,
+  cosmosSourceValidator: null,
+  validators: [],
+});
+
+const updateTransaction = (t, patch) => ({ ...t, ...patch });
+
+const redelegationStatusError = (a, t) => {
+  if (a.cosmosResources) {
+    const redelegations = a.cosmosResources.redelegations;
+
+    if (redelegations.length >= 7) {
+      return new CosmosTooMuchRedelegations();
```

Comment:
It shouldn't be a new error. please switch to `invariant` but expose us this `7` via an exported constant so our UI can takes this into account.

(UI should prevent it to happen before user see the error, so it's an invariant)